### DISCUSSION TITLE
How to set default values to a field of a record?

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#777] [Q&A]
Hello, I want some values to be set by default. How do I?

**Comment by ganigeorgiev** [comment]
Depending what the defaults are, you can:

1. Set the defaults client-side when sending the request data.

2. Set the defaults server-side using a [before create/update request hook](https://pocketbase.io/docs/event-hooks/#record-hooks), eg:

    ```go
    app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordCreateEvent) error {
        if e.Record.Collection().Name != "example" {
            return nil // not an "example" record
        }

        // set default value
        if e.Record.GetStringDataValue("yourField") == "" {
            e.Record.SetDataValue("yourField", "test")
        }

        return nil
    })
    ```
    (_In the upcoming v0.8 the recod accessor methods like `GetStringDataValue` and the setters like `SetDataValue` will be simplified to `GetString()` and `Set()`_)
